### ISSUE TITLE
Installation issue: gmsh

### ISSUE THREAD

**Issue by cessenat** [LLNL/spack#26408]
### Steps to reproduce the issue

```
$ spack install gmsh ^openblas ^openmpi %gcc@10.3.0

=> Installing fltk-1.3.3-ho3ibyy7ye36opuutfjdodljwr3t77pk
==> No binary for fltk-1.3.3-ho3ibyy7ye36opuutfjdodljwr3t77pk found: installing from source
==> Fetching https://mirror.spack.io/_source-cache/archive/f8/f8398d98d7221d40e77bc7b19e761adaf2f1ef8bb0c30eceb7beb4f2273d0d97.tar.gz
==> Applied patch /home/cessenat/spack/var/spack/repos/builtin/packages/fltk/font.patch
==> Applied patch /home/cessenat/spack/var/spack/repos/builtin/packages/fltk/fix_compare_val.patch
==> Applied patch /home/cessenat/spack/var/spack/repos/builtin/packages/fltk/type_cast.patch
==> Ran patch() for fltk
==> fltk: Executing phase: 'install'
==> fltk: Successfully installed fltk-1.3.3-ho3ibyy7ye36opuutfjdodljwr3t77pk
  Fetch: 5.12s.  Build: 1m 6.69s.  Total: 1m 11.81s.
[+] /home/cessenat/spack/opt/spack/linux-ubuntu21.04-x86_64_v2/gcc-10.3.0/fltk-1.3.3-ho3ibyy7ye36opuutfjdodljwr3t77pk
==> Installing gmsh-4.8.4-dyc3aomfbb4iajqa6wj6gbkv36bt22b7
==> No binary for gmsh-4.8.4-dyc3aomfbb4iajqa6wj6gbkv36bt22b7 found: installing from source
==> Fetching https://mirror.spack.io/_source-cache/archive/76/760dbdc072eaa3c82d066c5ba3b06eacdd3304eb2a97373fe4ada9509f0b6ace.tgz
==> No patches needed for gmsh
==> gmsh: Executing phase: 'cmake'
==> gmsh: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j2'

8 errors found in build log:
     1679    [ 95%] Linking CXX executable t9
     1680    /home/cessenat/spack/opt/spack/linux-ubuntu21.04-x86_64_v2/gcc-10.3.0/cmake-3.21.3-36k3lwt4
             lt6otzgzhbyy5ozclcaktwgr/bin/cmake -E cmake_link_script CMakeFiles/t9.dir/link.txt --verbos
             e=1
     1681    /home/cessenat/spack/lib/spack/env/gcc/g++ -O2 -g -DNDEBUG -rdynamic CMakeFiles/t9.dir/tuto
             rial/c++/t9.cpp.o -o t9  -Wl,-rpath,/tmp/cessenat/spack-stage/spack-stage-gmsh-4.8.4-dyc3ao
             mfbb4iajqa6wj6gbkv36bt22b7/spack-build-dyc3aom:/home/cessenat/spack/opt/spack/linux-ubuntu2
             1.04-x86_64_v2/gcc-10.3.0/gmp-6.2.1-kmvgnnl7n6kmrbwb4iejoqthfl4kqgmk/lib:/home/cessenat/spa
             ck/opt/spack/linux-ubuntu21.04-x86_64_v2/gcc-10.3.0/mmg-5.5.2-t5w5i7njejurht2lit7iauikxve4r
             rio/lib:/home/cessenat/spack/opt/spack/linux-ubuntu21.04-x86_64_v2/gcc-10.3.0/cgns-4.2.0-5x
             6bss36ipawp2mzu6fnytq6hpaorce7/lib:/home/cessenat/spack/opt/spack/linux-ubuntu21.04-x86_64_
             v2/gcc-10.3.0/hdf5-1.10.7-3yil2ograiogsnp6fwggdid5xepa4lyq/lib:/home/cessenat/spack/opt/spa
             ck/linux-ubuntu21.04-x86_64_v2/gcc-10.3.0/mesa-21.2.1-tr3qa73kcyybletytopcr5imdolh4qlt/lib 
             libgmsh.so.4.8.4 -L/home/cessenat/spack/opt/spack/linux-ubuntu21.04-x86_64_v2/gcc-10.3.0/fl
             tk-1.3.3-ho3ibyy7ye36opuutfjdodljwr3t77pk/lib -Wl,-rpath,/home/cessenat/spack/opt/spack/lin
             ux-ubuntu21.04-x86_64_v2/gcc-10.3.0/fltk-1.3.3-ho3ibyy7ye36opuutfjdodljwr3t77pk/lib -lfltk_
             images -lfltk_png -lfltk_z -lfltk_jpeg -lfltk_gl -lGLU -lGL -lfltk -lXext -lpthread -ldl -l
             m -lX11 /home/cessenat/spack/opt/spack/linux-ubuntu21.04-x86_64_v2/gcc-10.3.0/gmp-6.2.1-kmv
             gnnl7n6kmrbwb4iejoqthfl4kqgmk/lib/libgmp.so /home/cessenat/spack/opt/spack/linux-ubuntu21.0
             4-x86_64_v2/gcc-10.3.0/mmg-5.5.2-t5w5i7njejurht2lit7iauikxve4rrio/lib/libmmg.so /home/cesse
             nat/spack/opt/spack/linux-ubuntu21.04-x86_64_v2/gcc-10.3.0/med-4.0.0-r3keblhdneidv3algsad3d
             i2fnbaztoj/lib/libmedC.a /home/cessenat/spack/opt/spack/linux-ubuntu21.04-x86_64_v2/gcc-10.
             3.0/cgns-4.2.0-5x6bss36ipawp2mzu6fnytq6hpaorce7/lib/libcgns.so /home/cessenat/spack/opt/spa
             ck/linux-ubuntu21.04-x86_64_v2/gcc-10.3.0/hdf5-1.10.7-3yil2ograiogsnp6fwggdid5xepa4lyq/lib/
             libhdf5.so.103.3.0 /usr/lib/x86_64-linux-gnu/libsz.so -ldl /home/cessenat/spack/opt/spack/l
             inux-ubuntu21.04-x86_64_v2/gcc-10.3.0/mesa-21.2.1-tr3qa73kcyybletytopcr5imdolh4qlt/lib/libG
             L.so /usr/lib/x86_64-linux-gnu/libGLU.so -L/home/cessenat/spack/opt/spack/linux-ubuntu21.04
             -x86_64_v2/gcc-10.3.0/openblas-0.3.17-65xx5tikdbe6preumsddancentravxei/lib -lopenblas
     1682    [ 95%] Linking CXX executable t8
     1683    /home/cessenat/spack/opt/spack/linux-ubuntu21.04-x86_64_v2/gcc-10.3.0/cmake-3.21.3-36k3lwt4
             lt6otzgzhbyy5ozclcaktwgr/bin/cmake -E cmake_link_script CMakeFiles/t8.dir/link.txt --verbos
             e=1
     1684    /home/cessenat/spack/lib/spack/env/gcc/g++ -O2 -g -DNDEBUG -rdynamic CMakeFiles/t8.dir/tuto
             rial/c++/t8.cpp.o -o t8  -Wl,-rpath,/tmp/cessenat/spack-stage/spack-stage-gmsh-4.8.4-dyc3ao
             mfbb4iajqa6wj6gbkv36bt22b7/spack-build-dyc3aom:/home/cessenat/spack/opt/spack/linux-ubuntu2
             1.04-x86_64_v2/gcc-10.3.0/gmp-6.2.1-kmvgnnl7n6kmrbwb4iejoqthfl4kqgmk/lib:/home/cessenat/spa
             ck/opt/spack/linux-ubuntu21.04-x86_64_v2/gcc-10.3.0/mmg-5.5.2-t5w5i7njejurht2lit7iauikxve4r
             rio/lib:/home/cessenat/spack/opt/spack/linux-ubuntu21.04-x86_64_v2/gcc-10.3.0/cgns-4.2.0-5x
             6bss36ipawp2mzu6fnytq6hpaorce7/lib:/home/cessenat/spack/opt/spack/linux-ubuntu21.04-x86_64_
             v2/gcc-10.3.0/hdf5-1.10.7-3yil2ograiogsnp6fwggdid5xepa4lyq/lib:/home/cessenat/spack/opt/spa
             ck/linux-ubuntu21.04-x86_64_v2/gcc-10.3.0/mesa-21.2.1-tr3qa73kcyybletytopcr5imdolh4qlt/lib 
             libgmsh.so.4.8.4 -L/home/cessenat/spack/opt/spack/linux-ubuntu21.04-x86_64_v2/gcc-10.3.0/fl
             tk-1.3.3-ho3ibyy7ye36opuutfjdodljwr3t77pk/lib -Wl,-rpath,/home/cessenat/spack/opt/spack/lin
             ux-ubuntu21.04-x86_64_v2/gcc-10.3.0/fltk-1.3.3-ho3ibyy7ye36opuutfjdodljwr3t77pk/lib -lfltk_
             images -lfltk_png -lfltk_z -lfltk_jpeg -lfltk_gl -lGLU -lGL -lfltk -lXext -lpthread -ldl -l
             m -lX11 /home/cessenat/spack/opt/spack/linux-ubuntu21.04-x86_64_v2/gcc-10.3.0/gmp-6.2.1-kmv
             gnnl7n6kmrbwb4iejoqthfl4kqgmk/lib/libgmp.so /home/cessenat/spack/opt/spack/linux-ubuntu21.0
             4-x86_64_v2/gcc-10.3.0/mmg-5.5.2-t5w5i7njejurht2lit7iauikxve4rrio/lib/libmmg.so /home/cesse
             nat/spack/opt/spack/linux-ubuntu21.04-x86_64_v2/gcc-10.3.0/med-4.0.0-r3keblhdneidv3algsad3d
             i2fnbaztoj/lib/libmedC.a /home/cessenat/spack/opt/spack/linux-ubuntu21.04-x86_64_v2/gcc-10.
             3.0/cgns-4.2.0-5x6bss36ipawp2mzu6fnytq6hpaorce7/lib/libcgns.so /home/cessenat/spack/opt/spa
             ck/linux-ubuntu21.04-x86_64_v2/gcc-10.3.0/hdf5-1.10.7-3yil2ograiogsnp6fwggdid5xepa4lyq/lib/
             libhdf5.so.103.3.0 /usr/lib/x86_64-linux-gnu/libsz.so -ldl /home/cessenat/spack/opt/spack/l
             inux-ubuntu21.04-x86_64_v2/gcc-10.3.0/mesa-21.2.1-tr3qa73kcyybletytopcr5imdolh4qlt/lib/libG
             L.so /usr/lib/x86_64-linux-gnu/libGLU.so -L/home/cessenat/spack/opt/spack/linux-ubuntu21.04
             -x86_64_v2/gcc-10.3.0/openblas-0.3.17-65xx5tikdbe6preumsddancentravxei/lib -lopenblas
  >> 1685    /usr/bin/ld: /home/cessenat/spack/opt/spack/linux-ubuntu21.04-x86_64_v2/gcc-10.3.0/fltk-1.3
             .3-ho3ibyy7ye36opuutfjdodljwr3t77pk/lib/libfltk_gl.so: undefined reference to `XGetUtf8Font
             AndGlyph'
  >> 1686    /usr/bin/ld: /home/cessenat/spack/opt/spack/linux-ubuntu21.04-x86_64_v2/gcc-10.3.0/fltk-1.3
             .3-ho3ibyy7ye36opuutfjdodljwr3t77pk/lib/libfltk_gl.so: undefined reference to `XGetUtf8Font
             AndGlyph'
  >> 1687    collect2: error: ld returned 1 exit status
  >> 1688    make[2]: *** [CMakeFiles/t9.dir/build.make:109: t9] Error 1
     1689    make[2]: Leaving directory '/tmp/cessenat/spack-stage/spack-stage-gmsh-4.8.4-dyc3aomfbb4iaj
             qa6wj6gbkv36bt22b7/spack-build-dyc3aom'
  >> 1690    make[1]: *** [CMakeFiles/Makefile2:741: CMakeFiles/t9.dir/all] Error 2
     1691    make[1]: *** Waiting for unfinished jobs....
  >> 1692    collect2: error: ld returned 1 exit status
  >> 1693    make[2]: *** [CMakeFiles/t8.dir/build.make:109: t8] Error 1
     1694    make[2]: Leaving directory '/tmp/cessenat/spack-stage/spack-stage-gmsh-4.8.4-dyc3aomfbb4iaj
             qa6wj6gbkv36bt22b7/spack-build-dyc3aom'
  >> 1695    make[1]: *** [CMakeFiles/Makefile2:793: CMakeFiles/t8.dir/all] Error 2
     1696    make[1]: Leaving directory '/tmp/cessenat/spack-stage/spack-stage-gmsh-4.8.4-dyc3aomfbb4iaj
             qa6wj6gbkv36bt22b7/spack-build-dyc3aom'
     1697    make: *** [Makefile:169: all] Error 2

See build log for details:
  /tmp/cessenat/spack-stage/spack-stage-gmsh-4.8.4-dyc3aomfbb4iajqa6wj6gbkv36bt22b7/spack-build-out.txt

```

### Information on your system

$ spack debug report
* **Spack:** 0.16.2-4553-b96d836a72
* **Python:** 3.9.5
* **Platform:** linux-ubuntu21.04-x86_64_v2
* **Concretizer:** clingo

### Additional information

[spack-build-env.txt](https://github.com/spack/spack/files/7267724/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/7267725/spack-build-out.txt)

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by cessenat** [comment]
This is likely to be the same issue as encountered in 
https://github.com/spack/spack/issues/26051#issuecomment-932019667
(after solving med HDF5+MPI initial problem solved by PR https://github.com/spack/spack/pull/25312)

**Comment by cessenat** [comment]
@maruthinh you could try, as an alpha test:
spack install --no-checksum gmsh ^fltk@1.3.7

**Comment by cessenat** [comment]
This is solved by PR #26423
